src/validation.cpp

```diff
@@ -2517,6 +2523,27 @@ static bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockInd
 
         nInputs += tx.vin.size();
 
+        if (tx.IsCoinBase() && AreCoinbaseCheckAssetsDeployed()) {
+            for (unsigned int i = 0; i < tx.vout.size(); i++) {
```

Comment:
Why not use 
`for (auto vout : tx.vout)` Instead here.

Reply:
Updated to the for each loop